salt/modules/virt.py

```diff
@@ -2007,10 +2004,11 @@ def _handle_efi_param(boot, desc):
     elif type(efi_value) == bool and os_attrib == {}:
         if efi_value is True and parent_tag.find("loader") is None:
             parent_tag.set("firmware", "efi")
+            return True
         if efi_value is False and parent_tag.find("loader") is not None:
             parent_tag.remove(parent_tag.find("loader"))
             parent_tag.remove(parent_tag.find("nvram"))
-        return True
+            return True
```

Comment:
**suggestion (non-blocking)**: I'd love if we could refactor this function a bit. The bool logic smells funny (`is False` and `is True` is a code smell), and we can do `parent_tag.find("loader")` up to 3 times in here.

I took a stab on the test clinic to try and come up with a reasonable refactor but I'm not sure I grok enough about it.

Anyway, I don't think it should block this PR at all, but it'd be awesome if it could get simplified at some point!

Reply:
I agree this one would need a refactoring. I'll try to come to it later, but not in this PR.